blog/2021-08-14-react-navigation-6.0.md

```diff
@@ -0,0 +1,159 @@
+
+We also export a [`Header`](/docs/elements#header) component in the new elements library to use anywhere in your components.
+
+### Native navigation by default
+
+Historically, React Navigation has been mostly JS based, with animations and gestures written in JavaScript. While this works for a lot of apps, apps with heavy screens can suffer from poor performance. So we also wanted to address this by using native primitives for navigation.
+
+With React Navigation 5, we introduced [`@react-navigation/native-stack`](/docs/native-stack-navigator) package powered by [`react-native-screens`](https://github.com/software-mansion/react-native-screens), as well as a native backend for [`@react-navigation/material-top-tabs`](/docs/material-top-tab-navigator) powered by [`react-native-pager-view`](https://github.com/callstack/react-native-pager-view).
+
+In React Navigation 6, we made `@react-navigation/native-stack` the default choice for setting up Stack navigation. It uses UINavigationController on iOS and Fragments on Android to implement navigation natively. We also focused a lot on aligning the API of `@react-navigation/native-stack` with `@react-navigation/stack` so that it’d be easier to switch between them.
+
+Similarly, we switched `@react-navigation/material-top-tabs` to use `react-native-pager-view` by default.
+
+### Better type-safety
+
+React Navigation 5’s TypeScript support was much better than React Navigation 4. But some things such as `useNavigation` were still untyped by default. So we wanted to change that to make them more type-safe.
+
+In React Navigation 6, you don’t need to annotate `useNavigation` to get autocompletion and type checking. This is possible by defining a type for the screens globally using declaration merging:
+
+```js
+declare global {
+  namespace ReactNavigation {
+    interface RootParamList {
+      Home: undefined;
+      Profile: { userId: string };
+      NotFound: undefined;
+    }
+  }
+}
+```
+
+You can read [more about it in our TypeScript docs](/docs/typescript#specifying-default-types-for-usenavigation-link-ref-etc).
+
+### Flipper plugin
+
+We now have a [Flipper](https://fbflipper.com/) plugin. It includes similar functionality to the currently available Redux Devtools Extensions integration. You can see all navigation actions, and jump back n forth between earlier and new navigation states. In addition, it also includes a tab to test your linking configuration.
+
+Since the dev tools is built from scratch, we’re now free to add new features to make debugging easier in future.
+
+One advantage of the Flipper plugin over Redux Devtools Extension is that it doesn’t need Chrome Debugger to work. Since Chrome Debugger can sometimes affect and change how things work, we won’t have this problem with Flipper.
```

Comment:
```suggestion
One advantage of the Flipper plugin over Redux Devtools Extension is that it doesn’t need Chrome Debugger to work. Since Chrome Debugger can sometimes affect performance and even potentially change behavior, we think this is a more reliable solution.
```